docs/static/dead-letter-queues.asciidoc

```diff
@@ -0,0 +1,155 @@
+pipeline that uses the `dead_letter_queue` input plugin to read from the dead
+letter queue. The pipeline configuration that you use depends, of course, on
+what you need to do. For example, if the dead letter queue contains events that
+resulted from a mapping error in Elasticsearch, you can create a pipeline that
+reads the "dead" events, removes the field that caused the mapping issue, and
+re-indexes the clean events into Elasticsearch. 
+
+The following example shows how to read events from the dead letter queue and
+write the events to standard output: 
+
+[source,yaml]
+--------------------------------------------------------------------------------
+input {
+  dead_letter_queue {
+    path => "/path/to/data/dead_letter_queue" <1>
+    commit_offsets => true <2>
+    pipeline_id => "main" <3>
+  }
+}
+
+output {
+  stdout {
+    codec => rubydebug
+  }
+}
+--------------------------------------------------------------------------------
+
+<1> The path to the directory containing the dead letter queue. This is either
+the default, `path.data/dead_letter_queue`, or the value specified for
+`path.dead_letter_queue` in the `logstash.yml` file. 
+<2> When `true`, saves the offset. When the pipeline restarts, it will continue
+reading from the position where it left off rather than reprocessing all the
+items in the queue. You can set `commit_offsets` to `false` when you are
+exploring events in the dead letter queue and want to iterate over the events
+multiple times. 
+<3> The ID of the pipeline that's writing to the dead letter queue. The default
+is `"main"`.
+
+When the pipeline has finished processing all the events in the dead letter
+queue, it will continue to run and process new events as they stream into the
+queue. This means that you do not need to stop your production system to handle
+events in the dead letter queue. 
+
+When you read from the dead letter queue, you might not want to process all the
+events in the queue, especially if there are a lot of old events in the queue.
+You can start processing events at a specific point in the queue by using the
+`start_timestamp` option. This option configures the pipeline to start
+processing events based on the timestamp of when they entered the queue:
+
+[source,yaml]
+--------------------------------------------------------------------------------
+input {
+  dead_letter_queue {
+    path => "/path/to/data/dead_letter_queue" 
+    start_timestamp => 2017-06-06T23:40:37
+    pipeline_id => "main"
+  }
+}
+--------------------------------------------------------------------------------
+
+For this example, the pipeline starts reading all events that were delivered to
+the dead letter queue on or after June 6, 2017, at 23:40:37.
+
+//REVIEWERS: It's not clear to me what happens when the user configures start_timestamp and commit_offsets is true. If an offset's been committed, will the pipeline start reading at the offset, or go by the timestamp specified in start_timestamp.
+
```

Comment:
Yeah, this one is tricky. I think the `start_timestamp` should honor the committed offset and should restrict the range based on the timestamp after the commit point.

@robbavey you are right that the current behavior will confuse users. We should open an issue.